cli/cargo-ohrs/src/build/run.rs

```diff
@@ -93,53 +81,65 @@ pub fn build(c: Arc<RwLock<Context>>, arch: &Architecture) {
   let mut args = ctx.init_args.clone();
   args.extend([
     "--target",
-    &arch.target,
+    &arch.rust_target(),
     "--message-format=json-render-diagnostics",
   ]);
 
+  // respect cli extra args
+  args.extend(cargo_args.iter().map(|s| s.as_str()).collect::<Vec<_>>());
```

Comment:
`extend` usually takes `IntoIterator`, you should be able to drop `.collect::<Vec<_>>()` and avoid allocating it.